### PR TITLE
Refactor proctitle creation

### PR DESCRIPTION
**Pull request by mperham**
This allows other code to add elements to the proctitle.

### CODE REVIEW

File: test/test_manager.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def new_manager(opts)
         end
 
         it 'sets useful info to proctitle' do
-          assert_equal "sidekiq #{Sidekiq::VERSION} myapp [1 of 3 busy]", $0
+          assert_equal "sidekiq #{Sidekiq::VERSION} myapp [1 of 3 busy] ", $0
```
Comment (aprescott) [inline test/test_manager.rb]: Is it worth calling `strip` on the result so this extra space doesn't show up?

**Comment by mperham** [comment]
I knew someone would bring it up.  Dealt with it.  :sunglasses:

File: lib/sidekiq/manager.rb [hunk 2]
```diff
@@ -134,12 +134,18 @@ def real_thread(proxy_id, thr)
       @threads[proxy_id] = thr
     end
 
+    PROCTITLES = [
+      Proc.new { 'sidekiq'.freeze },
+      Proc.new { Sidekiq::VERSION },
+      Proc.new { |mgr, data| data['tag'] },
+      Proc.new { |mgr, data| "[#{mgr.busy.size} of #{data['concurrency']} busy]" },
+      Proc.new { |mgr, data| "stopping" if mgr.stopped? },
```
Comment (davydovanton) [inline lib/sidekiq/manager.rb]: it's style preference but may be will we use `proc` instead `Proc.new` and `x.(arg1, arg2)` instead `x.call(arg1, arg2)`?

Comment (mperham) [inline lib/sidekiq/manager.rb]: Please do!

Comment (davydovanton) [inline lib/sidekiq/manager.rb]: no problem :)

**Comment by davydovanton** [comment]
and may be we need to comment this or update wiki?

**Comment by mperham** [comment]
It's an internal API, not meant for public usage.  Not documented on purpose.

**Comment by davydovanton** [comment]
okay, I understand

**Pull request merged**